### PR TITLE
Mention saveLayer in the CustomPainter docs.

### LINKED ISSUE #11002: Explain when you should use saveLayer with a CustomPainter
Drawing with a `BlendMode` unfortunately doesn't give the expected results. In the following case the source `Rect` should be gone and the dest `Rect` (the red one) should be cut out in the bottom right corner. Instead the source `Rect` just gets black.

```dart
class TestPaint extends CustomPainter {
	@override
	void paint(Canvas canvas, Size size) {
		canvas.drawPaint(new Paint()..color = Colors.transparent);
		canvas.drawRect(
			new Rect.fromLTWH(50.0, 100.0, 200.0, 200.0),
			new Paint()..color = Colors.red
		);
		canvas.drawRect(
			new Rect.fromLTWH(150.0, 200.0, 200.0, 200.0),
			new Paint()..color = Colors.white
				..blendMode = BlendMode.dstOut
		);
	}
}
```
screenshot_20170627-222813

It doesn't depend on the color of the second `Rect`. Other `BlendMode`s like `srcIn` or `dstIn` don't seem to work either. Is this a bug?

### PR DESCRIPTION
**Pull request by Hixie**
Fixes https://github.com/flutter/flutter/issues/11002

### CODE REVIEW

File: packages/flutter/lib/src/rendering/custom_paint.dart [hunk 1]
```diff
@@ -119,6 +119,23 @@ typedef SemanticsBuilderCallback = List<CustomPainterSemantics> Function(Size si
...
+/// [CustomPainter]'s [Canvas] may be the same as that used by other widgets
+/// (including other [CustomPaint] widgets).
+///
+/// This is mostly unnoticeable, except when using unusual [BlendMode]s. For
+/// example, trying to use [BlendMode.dstOut] to "punch a hole" through a
+/// previously-drawn image may erase more that was intended, because previous
+/// widgets will have been painted onto the same canvas.
+///
+/// To avoid this issue, consider using [Canvas.saveLayer] and
+/// [Canvas.restoreLayer] when using such blend modes. Creating new layers is
```
Comment (goderbauer) [inline packages/flutter/lib/src/rendering/custom_paint.dart:135]: Isn't this just `Canvas.restore`?

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/rendering/custom_paint.dart [hunk 2]
```diff
@@ -119,6 +119,23 @@ typedef SemanticsBuilderCallback = List<CustomPainterSemantics> Function(Size si
...
+/// ## Composition and the sharing of canvases
+///
+/// Widgets (or rather, render objects) are composited together using a minimum
+/// number of [Canvas]es, for performance reasons. As a result, a
+/// [CustomPainter]'s [Canvas] may be the same as that used by other widgets
+/// (including other [CustomPaint] widgets).
+///
+/// This is mostly unnoticeable, except when using unusual [BlendMode]s. For
+/// example, trying to use [BlendMode.dstOut] to "punch a hole" through a
+/// previously-drawn image may erase more that was intended, because previous
```
[the remark this answers is not in the document]
Comment (Hixie) [reply packages/flutter/lib/src/rendering/custom_paint.dart:131]: fixed

File: packages/flutter/lib/src/rendering/custom_paint.dart (hunk 1 above)
Comment (Hixie) [reply packages/flutter/lib/src/rendering/custom_paint.dart:135]: oops yes

**Pull request merged**